vignettes/Deployment.Rmd

```diff
@@ -0,0 +1,79 @@
+---
+title: "Deployment"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Deployment}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include=FALSE}
+# Load packages.
+library(fontawesome)
+
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+
+To reap all the advantages of the `riskassessment` application, an organization needs to frequently execute and store results derived from within the app, including but not limited to:
+
+1. uploading new package info
+2. publishing reviewer comments
+3. storing metric weights
+4. making package usage decisions for GxP environments
+5. etc
+
+To accomplish this, the app automatically creates and populates a `SQLite` database, stored locally in the application's root directory when launched for the first time. In addition, a `loggit.json` file records some rudimentary logs of actions performed in the app and is also stored in the root directory. As you might expect, some deployment environments come with limitations that are not amicable to said storage method! Spoiler alert - shinyapps.io is one of them. This article will provide a run down of the following deployment options and supply helpful tips as it pertains to `riskassessment`:
+
+* shinyapps.io
+
+* RStudio Connect
+
+* Shiny Server
+
+* ShinyProxy
+
+Note! In the future, we'll consider adding utilities that would allow users to connect to an externa, or remote storage method, like a database. 
+
+
+<br>
+
+## shinyapps.io
+
+Beware: shinyapps.io **does not** offer persistent storage of `riskassessment`'s `SQLite` database or logs. Thus, it **is not** likely a viable deployment option for your organization or group. For example, our [demo application](https://rinpharma.shinyapps.io/riskassessment) (hosted on shinyapps.io) contains a pre-prepared database of packages that can't be permanently altered. It CAN be altered within a session, but any changes **will not** persist outside the session.
+
+
+<br>
+
+## Posit Connect & Shiny Server
+
+On a server, if you want to save files on disk, you'll need to set write permissions accordingly on the folder you want to save the database & logs.
+
+On Posit Connect, you need to use an absolute path to specify the directory where to save logs. You can find more information here: [Persistent Storage on Posit Connect](https://support.rstudio.com/hc/en-us/articles/360007981134-Persistent-Storage-on-RStudio-Connect).
+
+<br>
+
+## ShinyProxy
+
+With [ShinyProxy](https://www.shinyproxy.io/), you can use a Docker volume to write files outside of the application container. In `application.yml`, you use can something like this in the specs describing the application:
+
+> container-volumes: [ "/var/log/shinylogs:/root/logs" ]
+
+`/var/log/shinylogs` is a directory on the server where you deploy your applications with ShinyProxy. `/root/logs` is a directory inside your Docker image.
+
+
+
```

Comment:
Unsure exactly how to work this in, but I think it would be good to make some note about the utility of using the included `renv` for deployment. 

Exactly how to do this will depend on deployment modality -- e.g. for Connect, you would build `renv` locally, run `rsconnect::writeManifest`, and set up for git-backed deployment. 

The full scope of *how* to do it is likely outside of the scope of this vignette. However, you could mention something along the lines of "the application has been built and tested alongside a specific `renv`... be sure to deploy with this in mind...."